dbt/config.py

```diff
@@ -90,27 +90,94 @@ def colorize_output(config):
+    """A renderer provides configuration rendering for a given set of cli
+    variables and a render type.
+    """
+    def __init__(self, cli_vars):
+        self.context = {'env_var': env_var}
+        self.context['var'] = Var(None, self.context, cli_vars)
 
-    If the parsed entry is a string and has the name 'port', this will attempt
-    to cast it to an int, and on failure will return the parsed string.
+    @staticmethod
+    def _is_hook_path(keypath):
+        if not keypath:
+            return False
 
-    :param key str: The key to convert on.
-    :param value Any: The value to potentially render
-    :param ctx dict: The context dictionary, mapping function names to
-        functions that take a str and return a value
-    :return Any: The rendered entry.
-    """
-    if not isinstance(value, compat.basestring):
+        first = keypath[0]
+        # run hooks
+        if first in {'on-run-start', 'on-run-end'}:
+            return True
+        # model hooks
+        if first in {'seeds', 'models'}:
+            if 'pre-hook' in keypath or 'post-hook' in keypath:
+                return True
+
+        return False
+
+    @staticmethod
+    def _is_port_path(keypath):
+        return len(keypath) == 2 and keypath[-1] == 'port'
+
+    @staticmethod
+    def _convert_port(value, keypath):
+
+        if len(keypath) != 4:
+            return value
+
+        if keypath[-1] == 'port' and keypath[1] == 'outputs':
+            try:
+                return int(value)
+            except ValueError:
+                pass  # let the validator or connection handle this
         return value
-    result = dbt.clients.jinja.get_rendered(value, ctx)
-    if key == 'port':
-        try:
-            return int(result)
-        except ValueError:
-            pass  # let the validator or connection handle this
-    return result
+
+    def _render_project_entry(self, value, keypath):
+        """Render an entry, in case it's jinja. This is meant to be passed to
+        dbt.utils.deep_map.
+
+        If the parsed entry is a string and has the name 'port', this will
+        attempt to cast it to an int, and on failure will return the parsed
+        string.
+
+        :param value Any: The value to potentially render
+        :param key str: The key to convert on.
+        :return Any: The rendered entry.
+        """
+        # hooks should be treated as raw sql, they'll get rendered later
+        if self._is_hook_path(keypath):
+            return value
+
+        return self.render_value(value)
+
+    def render_value(self, value, keypath=None):
+        # keypath is ignored.
+        # if it wasn't read as a string, ignore it
+        if not isinstance(value, compat.basestring):
+            return value
+
+        return dbt.clients.jinja.get_rendered(value, self.context)
+
+    def _render_profile_data(self, value, keypath):
+        result = self.render_value(value)
+        if len(keypath) == 1 and keypath[-1] == 'port':
```

Comment:
same comment from `_convert_port`: someday profile-specific rendering info should go on the adapter

Reply:
Adapters will have to provide their own credentials descriptions when they're plugins so I imagine I'll be forced to deal with this then.